### PR TITLE
Feature/iterate videos

### PR DESCRIPTION
**Pull request by willread**
## Description
This change adds the core functionality of the `rise-video` component.

## Motivation and Context
Implements playback of single and multiple videos as described in [this section of the Video Epic Execution Strategy](https://docs.google.com/document/d/1LG_h_jQX8UlGLHfmZhEch0DCQyeVxmCnnDAUyU0CaoQ/edit#heading=h.wm9m8arbfz3p).

- Added a `rise-video-player` component which encapsulates the main video playback functionality. This is largely ported from `player-vjs` in the existing `widget-video` component
- Integrated the videojs library by a combination of concatenating the videojs and videojs-playlist js files to the component js bundle in an npm postbuild step, and converting the `videojs` css to a polymer `dom-module` which can be imported. This is non-optimal and will be revisited in a later part of the Video Epic
- Tweaked the main `rise-video` component to allow it to properly detect and handle changes to the `files` property at runtime, such as when customizing the component in the Template Editor.

## How Has This Been Tested?
- Both unit and integration tests have been added for all new features
- Continuous testing during development using a custom template in Rise Chrome Player running as a Chrome extension via Charles Proxy
- A build of the component is available in [this template](https://widgets.risevision.com/staging/pages/2019.07.24.00.00/rise-video/src/template.html) and [this presentation](https://apps.risevision.com/templates/edit/3dfa38f2-7b70-4abe-858e-4cb14f81d139/?cid=7fa5ee92-7deb-450b-a8d5-e5ed648c575f) and has been tested in the Rise Chrome Player running as a Chrome extension

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
    - Manual testing completed
    - Automated tests added
    - No monitoring issues, except where noted in Release Checklist Items Skipped below
    - Rollback is simple (revert to release v1.0.0), and component is not used outside of test templates so chance of impact is minimal
    - Documentation has been checked and should still be accurate with these changes
    - Internal Communication with Support not necessary, as this component isn't used outside of test templates
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
- I won't be available over the weekend but since this component isn't used yet chance of impact is minimal

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "description": "Web component for playing video files on a Rise Vision Template page",
   "scripts": {
     "prebuild": "eslint .",
+    "postbuild": "concat -o dist/rise-video.js dist/rise-video.js node_modules/video.js/dist/video.min.js node_modules/videojs-playlist/dist/videojs-playlist.min.js",
```
Comment (santiagonoguez) [inline package.json]: @willread this is the temporary solution for packaging, right ?

Comment (willread) [reply package.json]: @santiagonoguez Yes! I'll definitely be revisiting this later in the epic.

File: src/rise-video-player.js [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        placeholderHeight: placeholderBounds.height,
+        placeholderAspect: getAspectRatio( placeholderBounds.width, placeholderBounds.height ),
+        videoWidth: this._playerInstance.videoWidth(),
+        videoHeight: this._playerInstance.videoHeight(),
+        videoAspect: getAspectRatio( this._playerInstance.videoWidth(), this._playerInstance.videoHeight() ),
+        fileUrl: this._playerInstance.currentSrc()
+      }
+    };
+    
+    // TODO: Log aspect event
```
Comment (santiagonoguez) [inline src/rise-video-player.js]: @willread let's ask later Stu if we'll need this aspect info to go to BQ later. If so, maybe with the flag that logs only once per day.

File: src/utils.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  return type;
+}
+
+export function getAspectRatio( width, height ) {
+
+  var r;
+
+  function gcd( a, b ) {
```
Comment (santiagonoguez) [inline src/utils.js]: @willread  r and gcd comes from old code ? I guess we could come with better names now, thanks !

Comment (willread) [reply src/utils.js]: @santiagonoguez I'll tidy that up and add some quick tests for these utility functions as well.

Comment (willread) [reply src/utils.js]: @santiagonoguez Pushed some changes to the utility functions and wrote some tests for them.

Comment (willread) [reply src/utils.js]: @santiagonoguez I tidied up those utility functions and added some tests

Comment (santiagonoguez) [reply src/utils.js]: cool, thanks !

**Pull request merged**